analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -448,121 +425,98 @@ def get_query(source: ObjectStream) -> ObjectStream:
 
 
 # %% [markdown]
-# ### Standalone ServiceX for subsequent `coffea` processing
+# ### Caching the queried datasets with `ServiceX`
 #
-# Using `servicex-databinder`, we can execute a query and download the output.
-# As the files are currently accessible through `rucio` only with ATLAS credentials, you need to use an ATLAS ServiceX instance to run this (for example via the UChicago coffea-casa analysis facility).
+# Using the queries created with `func_adl`, we are using `ServiceX` to read the ATLAS Open Data files to build cached files with only the specific event information as dictated by the query.
+
+# %%
+if USE_SERVICEX:
+    
+    # dummy dataset on which to generate the query
+    dummy_ds = ServiceXSourceUpROOT("cernopendata://dummy", "events", backend_name="uproot")
+
+    # tell low-level infrastructure not to contact ServiceX yet, only to
+    # return the qastle string it would have sent
+    dummy_ds.return_qastle = True
+
+    # create the query
+    query = get_query(dummy_ds)
+    query = query.value()
```

Comment:
Might as well do this in one line
```python
query = get_query(dummy_ds).value()
```
as I'm not aware of anything else we'd do with this intermediate object.

Reply:
[b1561e3](https://github.com/iris-hep/analysis-grand-challenge/pull/103/commits/b1561e3a281d83fb5803928d8e6c5d46502fb609)